doc/contributing.md

```diff
@@ -479,6 +482,74 @@ make check   # Runs black, blackdoc, docformatter, flake8 and isort (in check mo
+2. Open a ['Wrap `<module-name>`' initial feature implementation PR](#initial-feature-implementation).
+3. Open an ['Add missing aliases to `<module-name>`' documentation PR](#add-missing-aliases).
+4. Open a 'support additional functionality in module' PR (optional).
+5. Add ['Gallery example for module' documentation PR](#contributing-gallery-plots).
+6. Add ['Tutorial for module' documentation PR](#contributing-tutorials) (optional).
+
+These steps will be tracked in the 'Wrapper for `<module-name>`' issue and the
+['wrapping GMT modules'](https://github.com/GenericMappingTools/pygmt/projects/9)
+project board. The pull requests can be split between multiple contributors and
+there is no obligation for a single contributor to complete all steps. Please
+comment on the initial 'Wrapper for `<module-name>`' if you would like to open
+a pull request for any of these tasks to avoid redundant efforts.
+
+#### Request wrapper for GMT module
+
+* Find the [*Issues*](https://github.com/GenericMappingTools/pygmt/issues) tab on the
+  top of the GitHub repository and click *New Issue*.
+* Click on *Get started* next to *Feature request - Wrap new GMT module*.
+* Follow the prompts for filling out the issue template.
+
+#### Initial feature implementation
+
+First, comment on the 'Wrapper for `<module-name>`' issue that you will be
+working on the initial feature implementation. This first pull request should
+be as minimal as possible - only adding the required functionality (i.e.,
+wrapping the required GMT arguments and supporting the primary input/output
+types).
+
+The following steps are common to all initial implementation pull requests that
+wrap a new GMT module (e.g., [initial grdfill implementation](https://github.com/GenericMappingTools/pygmt/pull/1276/files)):
+
+* Create a new module `<module-name>.py` in `pygmt/src`. The module docstring
+  should include the module name and a short description of the functionality
+  (e.g., `grdfill - Fill blank areas from a grid.`).
+* Add a function `<module-name>` to the module. When writing the new function,
+  it is generally easiest to reference the source code for other functions that
+  input/output similar object types.
+* Add a detailed docstring following the [numpy style guide](https://numpydoc.readthedocs.io/en/latest/format.html).
+* Add the function to the import statements in `pygmt/src/__init__.py`.
+* Add the function to the import statements in `pygmt/__init__.py`.
+* Add the function to appropriate section of the API documentation in `doc/api/index.rst`.
+* Add a testing module `test_<module-name>.py` in `pygmt/tests`, following
+  the guidelines in the [testing your code](#testing-your-code) section.
+
+#### Add missing aliases
+
+After the initial implementation, the missing aliases can be added in a
+separate PR (e.g., [add missing aliases to grd2xyz](https://github.com/GenericMappingTools/pygmt/pull/1537/files)).
```

Comment:
Might need to get a newer example once we go with the long option standard in #1932.